test/unit/session/test_session.py

```diff
@@ -221,12 +221,23 @@ def test_box_session_seeks_file_after_retry(box_session, mock_network_layer, ser
     assert mock_file_2.seek.has_calls(call(3) * 2)
 
 
-def test_box_session_raises_for_non_json_response(box_session, mock_network_layer, non_json_response, test_url):
+def test_box_session_raises_for_non_json_response(box_session, mock_network_layer, non_json_response, generic_successful_response, test_url):
     # pylint:disable=redefined-outer-name
-    mock_network_layer.request.side_effect = [non_json_response]
+    mock_network_layer.request.side_effect = [non_json_response, generic_successful_response]
 
+    API.MAX_RETRY_ATTEMPTS = 1
+    box_session.get(url=test_url)
+    API.MAX_RETRY_ATTEMPTS = 5
+
+
+def test_box_session_raises_for_non_json_response_after_retry(box_session, mock_network_layer, non_json_response, test_url):
+    # pylint:disable=redefined-outer-name
+    mock_network_layer.request.side_effect = [non_json_response, non_json_response]
```

Comment:
[non_json_response] * API.MAX_RETRY_ATTEMPTS , and you don't have to modify this variable